src/api/spec/models/obs_factory/distribution_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe ObsFactory::Distribution do
+  describe '::new' do
+    def strategy_for(name)
+      ObsFactory::Distribution.new(create(:project, name: name)).strategy
+    end
+
+    it { expect(strategy_for('openSUSE:Factory')).to         be_kind_of ObsFactory::DistributionStrategyFactory }
+    it { expect(strategy_for('openSUSE:Factory:PowerPC')).to be_kind_of ObsFactory::DistributionStrategyFactoryPPC }
+    it { expect(strategy_for('openSUSE:42.3')).to            be_kind_of ObsFactory::DistributionStrategyOpenSUSE }
+    it { expect(strategy_for('SUSE:SLE-12-SP1:GA')).to       be_kind_of ObsFactory::DistributionStrategySLE12SP1 }
+    it { expect(strategy_for('SUSE:SLE-15:GA')).to           be_kind_of ObsFactory::DistributionStrategySLE15 }
+    it { expect(strategy_for('SUSE:SLE-15-SP1:GA')).to       be_kind_of ObsFactory::DistributionStrategySLE15 }
+    it { expect(strategy_for('SUSE:SLE-12-SP4CASP1')).to     be_kind_of ObsFactory::DistributionStrategyCasp }
```

Comment:
Aehm, that's not casp :)

The project name is more like SUSE:SLE-12-SP3:Update:Products:CASP20

but I wonder if we couldn't put the strategy into an obs attribute (to be edited by the admin only).

Reply:
\*frowns\*
You start to like the OBS attribute system?^^

Reply:
I like databases in general :)

Reply:
Well, the test is just about whether the matches are correct. The regex is quite relaxed as it seems. Anyhow, I changed the project string now.